lib/executor.js

```diff
@@ -175,6 +175,7 @@ function setupDataSources(app, instructions) {
       useEnvVars: true,
     };
     obj = getUpdatedConfigObject(app, obj, opts);
+    obj.lazyConnect = true;
```

Comment:
@raymondfeng I think my previous answer is for "when connect will be called" but not "when connect will be triggered in initialize", so for example if user create a datasource from boot script, it will be triggered, since `lazyConnect` is undefined.
```
var myDS = new DataSource(dsName, dsConfig);
// will trigger connect in initialize
```